WooCommerce/Classes/ViewRelated/Orders/Order Details/OrderDetailsViewController.swift

```diff
@@ -704,9 +704,15 @@ private extension OrderDetailsViewController {
     }
 
     func editCustomerNoteTapped() {
-        let viewModel = EditCustomerNoteViewModel(order: viewModel.order)
-        let editNoteViewController = EditCustomerNoteHostingController(viewModel: viewModel)
-        present(editNoteViewController, animated: true, completion: nil)
+        let editNoteViewController = EditCustomerNoteHostingController(viewModel: viewModel.editNoteViewModel)
+
+        /// This notice presenter is needed because if optimistic updates are not
+        /// enabled the modal is not dismissed upon failure.
+        let errorNoticePresenter = DefaultNoticePresenter()
+        errorNoticePresenter.presentingViewController = editNoteViewController
+        viewModel.editNoteViewModel.modalNoticePresenter = errorNoticePresenter
```

Comment:
Super nit: we could probably move this part to the initializer of `EditCustomerNoteHostingController` to simplify the setup here.

Reply:
much better 🙂 Thanks